runtime/pkg/formatter/count_digits.go

```diff
@@ -0,0 +1,15 @@
+package formatter
+
+import "regexp"
+
+// countDigits counts all numeric digits in a string.
+func countDigits(numStr string) int {
+	re := regexp.MustCompile(`\D`)
+	return len(re.ReplaceAllString(numStr, ""))
+}
+
+// countNonZeroDigits counts all non-zero numeric digits in a string.
+func countNonZeroDigits(numStr string) int {
+	re := regexp.MustCompile(`[^1-9]`)
+	return len(re.ReplaceAllString(numStr, ""))
+}
```

Comment:
If you move the `MustCompile` results into globals, it will only need to compile the regex strings once instead of on every invocation